src/cylinder.cpp

```diff
@@ -158,7 +158,25 @@ inline bool Cylinder::handleCollition(Walker& walker, Collision &colision, Vecto
         colision.bounced_direction = temp_step.normalized();
 
     }
-
+    double percolation_in_ex= percolation;
+    double percolation_ex_in= sqrt(D_in/D_ex)*percolation_in_ex;
```

Comment:
This implements the asymmetrical percolation coefficient from a specific model (it would be good to add it as a reference in the comments). 

However, it should be passed as an argument rather than to be the default percolation model. (The default is a constant probability), which can be changed as an user input to this specific model.